src/precice/impl/DataContext.cpp

```diff
@@ -84,40 +85,39 @@ bool DataContext::hasMapping() const
 
 bool DataContext::isMappingRequired()
 {
-  using namespace mapping;
   if (not hasMapping()) {
     return false;
   }
 
-  auto       timing    = _mappingContext.timing;
-  const bool hasMapped = _mappingContext.hasMappedData;
-  const bool mapNow    = (timing == MappingConfiguration::ON_ADVANCE) || (timing == MappingConfiguration::INITIAL);
-
-  if ((not mapNow) || hasMapped) {
-    return false;
-  }
-
-  return true;
+  return std::any_of(_mappingContext.begin(), _mappingContext.end(), [](const auto &context) {
+                const auto timing    = context.timing;
+                const bool mapNow    = (timing == mapping::MappingConfiguration::ON_ADVANCE) || (timing == mapping::MappingConfiguration::INITIAL);
+                return (mapNow && !context.hasMappedData); });
 }
 
 void DataContext::mapData()
 {
   PRECICE_ASSERT(hasMapping());
-  int fromDataID = getFromDataID();
-  int toDataID   = getToDataID();
-  _toData->toZero();
-  _mappingContext.mapping->map(fromDataID, toDataID);
-  PRECICE_DEBUG("Mapped values = {}", utils::previewRange(3, _toData->values()));
+  // Reset the toData
+  std::for_each(_toData.begin(), _toData.end(), [](auto &data) { data->toZero(); });
```

Comment:
`std::for_each` is nice, but since you already have a loop below, I would directly do the call there.

Reply:
Alternatively, you could also remove the loop and put everything into your lambda function like above with the `std::any_of` in `isMappingRequired`.

Reply:
I went for the first, as I need the index in the loop.